recipes-bsp/u-boot/u-boot_%.bbappend

```diff
@@ -6,7 +6,7 @@ SRC_URI:append:rpi = " \
 
 SRC_URI:append:rpi = " file://0001-rpi-always-set-fdt_addr-with-firmware-provided-FDT-address.patch"
 
-DEPENDS:append:rpi = " u-boot-default-script"
+DEPENDS:append:rpi = " ${@oe.utils.conditional('UBOOT_ENV', '', 'u-boot-default-script', '', d)}"
```

Comment:
I think its better to tweak `u-boot-default-scripts` recipe via a bbappend in such a distro.  This is introducing a variable that wont be used elsewhere.

Reply:
Hey @kraj UBOOT_ENV is not a new variable invented by us. From what I read in the code it's the "official" core way of adding a boot.scr in a standardized manner ... https://github.com/openembedded/openembedded-core/blob/master/meta/classes-recipe/uboot-config.bbclass#L61 ... the u-boot_%.bbappend of meta-raspberrypi breaks that behaviour right now.

Maybe we should also check if template support for this in openembedded-core would be ok.

Reply:
right I did not check that the variable is already in core layer. However, this variable is relevant for deploy tasks. the logic is still twisted. IMO this change seems more appropriate in u-boot-defaults-script recipe for meta-rpi since this recipe provides the UBOOT_ENV expected file.